slither/utils/expression_manipulations.py

```diff
@@ -91,43 +106,85 @@ def copy_expression(
         # (.. ? .. : ..).add
         if isinstance(expression, MemberAccess):
             next_expr = expression.expression
-            if self.apply_copy(next_expr, true_expression, false_expression, f_expression):
+            if self.conditional_not_ahead(
+                next_expr, true_expression, false_expression, f_expression
+            ):
                 self.copy_expression(
                     next_expr, true_expression.expression, false_expression.expression
                 )
-
+        # pylint: disable=too-many-nested-blocks
```

Comment:
Can we move part of the function's logic into internal functions?

This would reduce the code complexity, and allows us to remove this warning